source/databases-collections/run-command.txt

```diff
@@ -0,0 +1,184 @@
+.. _ruby-run-command:
+
+======================
+Run a Database Command
+======================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: administration, code example
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-short+} to
+run a database command. You can use database commands to perform a
+variety of administrative and diagnostic tasks, such as fetching server
+statistics, initializing a replica set, or running an aggregation pipeline.
+
+.. important:: Prefer Driver Methods to Database Commands
+
+   The driver provides wrapper methods for many database commands.
+   If possible, we recommend using these methods instead of executing
+   database commands.
+   
+   To perform administrative tasks, use the :mongosh:`MongoDB Shell </>`
+   instead of the {+driver-short+}. The shell provides helper methods
+   that might not be available in the driver.
+   
+   If there are no available helpers in the driver or the shell, you
+   can use the ``db.runCommand()`` shell method or the driver's
+   ``command`` method, which is described in this guide.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To access this database
+from your {+language+} application, create a ``Mongo::Client`` object that connects to an Atlas cluster
+and assign the following value to your ``database`` variable:
+
+.. literalinclude:: /includes/databases-collections/run-command.rb
+    :language: scala
+    :dedent:
+    :start-after: start-db
+    :end-before: end-db
+
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+Execute a Command
+-----------------
+
+To run a database command, run the ``command`` instance method of a ``Mongo::Database``
+instance and pass the name of the operation to run as a parameter.
+
+The following example calls the ``command`` method to run the ``hello`` command, which
+returns information about the server:
+
+.. literalinclude:: /includes/databases-collections/run-command.rb
+   :language: scala
+   :dedent:
+   :start-after: start-hello
+   :end-before: end-hello
+
+.. tip::
+
+   To view a full list of database commands and their corresponding
+   parameters, see :manual:`Database Commands </reference/command/>` in 
+   the {+mdb-server+} manual.
+
+Set a Read Preference
+----------------------
+
+The ``command`` method does not inherit the read preference you might
+have set on your ``Database`` instance. By default, ``command``
+uses the ``primary`` read preference.
+
+You can set a read preference for the command execution by passing a
+``ReadPreference`` instance as a parameter to ``command``, as
+shown in the following code:
```

Comment:
The example given after this paragraph does not use a ``ReadPreference`` instance, instead passing a hash of read preference options. Maybe the paragraph could be adjusted something like:

```suggestion
You can set a read preference for the command execution by passing the
``:read`` option to the ``command`` method, as shown in the following code:
```